### PR TITLE
BIOSYS-90 Ability for the API client to include or exclude fields in

### PR DESCRIPTION
**Pull request by serge-gaia**
…serialization

Most of the code is test

### CODE REVIEW

File: biosys/apps/main/api/serializers.py [hunk 1]
```diff
@@ -126,6 +126,16 @@ def __init__(self, instance=None, **kwargs):
         # from the species_naming_facade above.
         self.species_name_id_mapping_cached = None
```
Comment (serge-gaia) [inline biosys/apps/main/api/serializers.py:128]: That's the implementation.

File: biosys/apps/main/tests/api/helpers.py [hunk 2]
```diff
@@ -135,6 +135,63 @@ def _create_record(self, client, dataset, record_data):
         self.assertIsNotNone(record)
         return record
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/helpers.py:137]: test helpers to create dataset and records with one call

File: biosys/apps/main/tests/api/test_record_serialization.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    def test_only_geometry(self):
+        """
+        Scenario: a web map user needs only the geometry field.
+            Given some records with geometry are created
+            And I request a get 'dataset-record' with fields=geometry
+            Then it should return only the geometry field
+        """
+        # records are created in setup
+        client = self.custodian_1_client
+        url = reverse('api:dataset-records', kwargs={'pk': self.dataset.pk})
```
Comment (serge-gaia) [inline biosys/apps/main/tests/api/test_record_serialization.py:27]: query params

File: biosys/apps/main/api/serializers.py [hunk 4]
```diff
@@ -126,6 +126,16 @@ def __init__(self, instance=None, **kwargs):
...
 
+        # dynamic fields
+        request = ctx.get('request')
+        if request:
+            expected_fields = request.query_params.getlist('fields', [])
+            if expected_fields:
+                existing_fields = self.fields.keys()
+                excluded_fields = set(existing_fields) - set(expected_fields)
+                for field in excluded_fields:
+                    self.fields.pop(field)
```
Comment (tony-gaia) [inline biosys/apps/main/api/serializers.py:137]: could probably do this in one line

```python
self.fields = {k: v for k: v in self.fields.items() if k in expected_fields}
```

Comment (serge-gaia) [reply biosys/apps/main/api/serializers.py:137]: `self.fields` cannot be set. It's a getter property.

**Pull request merged**